### PR TITLE
Enable to add a note with each configuration histories

### PR DESCRIPTION
**Pull request by hassaku**
This PR enables to add a note with each configuration histories.
See the following screen capture.

fluentd

### CODE REVIEW

File: app/views/fluentd/settings/histories/_list.html.haml [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
-      = link_to(file.name, daemon_setting_history_path(id: file.file_id))
-      %label= t('terms.backup_time')
-      = file.ctime.strftime(I18n.t 'time.formats.default')
+    %tr
+      %td= link_to(file.name, daemon_setting_history_path(id: file.file_id))
+      %td= file.ctime.strftime t('time.formats.default')
+      %td
+        = form_for :note, url: daemon_setting_note_path(id: file.note.file_id), method: :patch do |f|
+          = f.text_field :content, value: file.note.content, class: "note-content", id: nil
+          = submit_tag t('terms.save'), class: 'btn'
```
Comment (uu59) [inline app/views/fluentd/settings/histories/_list.html.haml]: more needed `btn-default` class

File: spec/features/setting_spec.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
     page.should have_css('h1', text: I18n.t('fluentd.settings.show.page_title'))
     page.should have_link(I18n.t('terms.edit'))
     page.should have_css('pre', text: 'GREAT CONFIG HERE')
-    expect(all('.row li').count).to eq Settings.histories_count_in_preview #links to hisotries#show
+    expect(all('.row tr').count).to eq Settings.histories_count_in_preview + 1 # links to hisotries#show + 1 table header
```
Comment (uu59) [inline spec/features/setting_spec.rb]: :+1:

**Comment by yakitorii** [comment]
You should remove a related note file on deleting a backup file. 

https://github.com/fluent/fluentd-ui/blob/master/app/models/fluentd/agent/local_common.rb#L86

File: app/models/concerns/fluentd/setting_archive/archivable.rb [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+        def file_path_of(dir, id)
+          file_path = Pathname.new(dir).join("#{id}#{self::FILE_EXTENSION}")
+          raise "No such a file #{file_path}" unless File.exist?(file_path)
+          file_path
+        end
       end
 
       def file_id
-        @file_id ||= with_file { name.gsub(/.conf\Z/,'') }
+        @file_id ||= with_file { name.gsub(/#{self.class::FILE_EXTENSION}\Z/,'') }
```
Comment (uu59) [inline app/models/concerns/fluentd/setting_archive/archivable.rb]: should `Regexp.escape(self.class::FILE_EXTENSION)` as @kou commented.

File: app/views/fluentd/settings/histories/_list.html.haml (hunk 1 above)
Comment (hassaku) [inline app/views/fluentd/settings/histories/_list.html.haml]: Apply that class and adjust it with text_field.

File: app/models/concerns/fluentd/setting_archive/archivable.rb (hunk 3 above)
Comment (hassaku) [inline app/models/concerns/fluentd/setting_archive/archivable.rb]: I overlooked that :disappointed:

**Pull request merged**